core/kernel/persistence/smoothsql/class.Resource.php

```diff
@@ -655,4 +610,70 @@ public function getServiceLocator()
     {
         return $this->getModel()->getServiceLocator();
     }
+
+    /**
+     * @param core_kernel_classes_Resource $resource
+     * @param array $properties
+     * @param string $userIdentifier
+     * @param string $dataLanguage
+     * @return array
+     */
+    private function buildTrippleArray(
+        core_kernel_classes_Resource $resource,
+        array $properties,
+        string $userIdentifier,
+        string $dataLanguage
+    ) {
+        $triples = [];
+        foreach ($properties as $propertyUri => $value) {
+            $property = $this->getModel()->getProperty($propertyUri);
+
+            $lang = ($property->isLgDependent() ? $dataLanguage : '');
+
+            $formattedValues = $this->normalizePropertyValues($value);
+
+            foreach ($formattedValues as $object) {
```

Comment:
by splitting again this `foreach` inside a different function we will have shell function / wrapper that actually does nothing because the call is done to a different function anyways
`core_kernel_classes_Triple::createTriple`
thus making the code even more verbose 
so i think we are safe to keep in the current scope the foreach